<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Martin Mestdagh, Zacharie Ménétrier" />

<meta name="date" content="2019-04-24" />

<title>ReMapEnrich: advanced use</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">ReMapEnrich: advanced use</h1>
<h4 class="author">Martin Mestdagh, Zacharie Ménétrier</h4>
<h4 class="date">2019-04-24</h4>



<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>In this vignette, we will be discovering more advanced functions and possibilities of this package. You may want to read the basics functions first in order to understand the principles of enrhciment analysis.</p>
</div>
<div id="loading-a-catalogue" class="section level1">
<h1>Loading a catalogue</h1>
<p>You have a choice of either loading the ReMap catalogue, or loading your own catalogue of peaks/regions.</p>
<div id="download-the-remap-catalogue" class="section level2">
<h2>Download the ReMap catalogue</h2>
<p>There is 4 different versions of the ReMap catalogue. The current version of ReMap 2018 <span class="citation">[1]</span> and the previous ReMap 2015 release <span class="citation">[2]</span>. Both are declined in hg19 or hg38 assemblies. This function will download the files locally.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create a local directory for the tutorial</span>
demo.dir &lt;-<span class="st"> &quot;~/ReMapEnrich_demo&quot;</span>
<span class="kw">dir.create</span>(demo.dir, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)

<span class="co"># Use the function DowloadRemapCatalog</span>
remapCatalog2018hg38 &lt;-<span class="st"> </span><span class="kw">downloadRemapCatalog</span>(demo.dir)

<span class="co"># Or download other versions</span>
remapCatalog2015hg19 &lt;-<span class="st"> </span><span class="kw">downloadRemapCatalog</span>(demo.dir, <span class="dt">version =</span> <span class="st">&quot;2015&quot;</span>, <span class="dt">assembly =</span> <span class="st">&quot;hg19&quot;</span>)</code></pre></div>
<p>The genomic ranges object remapCatalog now contains the full ReMap 2015 catalogue<span class="citation">[2]</span>. A file has also been downloaded at “~/ReMapEnrich_demo/remap1_hg38_nrPeaks.bed” (0.544 GB when not compressed).</p>
</div>
<div id="load-the-remap-catalogue" class="section level2">
<h2>Load the ReMap catalogue</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load the ReMap catalogue and convert it to Genomic Ranges</span>
remapCatalog &lt;-<span class="st"> </span><span class="kw">bedToGranges</span>(remapCatalog2018hg38)</code></pre></div>
</div>
<div id="loading-a-custom-catalogue" class="section level2">
<h2>Loading a custom catalogue</h2>
<p>Another funtionnality of this package is the ability to load a custom catalogue instead of ReMap. This is quite simple starting from a BED file as your catalogue. In this example, we load a custom catalogue present in <code>extdata/</code>, this is just a BED file for chr22 as dummy example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load the ReMapEnrich library</span>
<span class="kw">library</span>(ReMapEnrich) 

<span class="co"># Load the example catalogue as a BED file.</span>
catalogFile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,
                                    <span class="st">&quot;ReMap_nrPeaks_public_chr22.bed&quot;</span>,
                                    <span class="dt">package =</span> <span class="st">&quot;ReMapEnrich&quot;</span>)
catalog &lt;-<span class="st"> </span><span class="kw">bedToGranges</span>(catalogFile)</code></pre></div>
</div>
</div>
<div id="download-encode-peaks" class="section level1">
<h1>Download ENCODE peaks</h1>
<p>We have added a fucntionnality to direclty fetch ENCODE peaks. Wkith ReMapEnrich tt is possible to download genomic regions of any ENCODE experiments with its ENCODE ID. In this example we download a BED file for the H3K27 histone mark in MCF-7 cell line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Downloading the ENCFF001VCU regions.</span>
ENCFF001VCU &lt;-<span class="st"> </span><span class="kw">bedToGranges</span>(<span class="kw">downloadEncodePeaks</span>(<span class="st">&quot;ENCFF001VCU&quot;</span>, demo.dir))</code></pre></div>
<p>The ENCFF001VCU variable now contains all the regions of the given experiment a <code>Grange</code> object. It is possible to use it in future enrichment analysis.</p>
</div>
<div id="compute-enrichment" class="section level1">
<h1>Compute enrichment</h1>
<p>The basic way to compute an enrichment is to run with default parameters. - no universe - single core - Default shuffling - defautl overlaps</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">enrichment &lt;-<span class="st"> </span><span class="kw">enrichment</span>(ENCFF001VCU, remapCatalog)</code></pre></div>
</div>
<div id="using-a-universe" class="section level1">
<h1>Using a universe</h1>
<p>Enrichment analysis often surestimate the p-values. Using a universe is setting constraints on the shuffling function, resulting in more reasonable probabilities. The universe is simply another set of genomic regions, in a <code>GeomicRanges</code> format, that will prevent shuffles to take place outside of it. Using a universe will reduce the analysis to certain portions of the genome. However the ReMapEnrich package can also work without a universe, in this case it will use the entire genome.</p>
<div id="choosing-an-universe" class="section level2">
<h2>Choosing an universe</h2>
<p>One of the most diffcult task in using a universe is choosing one that is relevant for you data. No methods are perfect, once you understand what is the purpose of using a universe, it is be up to you to specify which one to use in order to accomplish a specific analysis. But in general, the universe could be understood as <strong>the set of regions that could have been used as query</strong>.</p>
</div>
<div id="enrichment-using-a-universe" class="section level2">
<h2>Enrichment using a universe</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Download a universe.</span>
universe &lt;-<span class="st"> </span><span class="kw">bedToGranges</span>(<span class="kw">downloadEncodePeaks</span>(<span class="st">&quot;ENCFF718QVA&quot;</span>, demo.dir))
<span class="co"># Convert ReMap to GRanges</span>
remapCatalog &lt;-<span class="st"> </span><span class="kw">bedToGranges</span>(remapCatalog2018hg38)
<span class="co"># Create the enrichment with the universe.</span>
enrichment &lt;-<span class="st"> </span><span class="kw">enrichment</span>(ENCFF001VCU, remapCatalog, universe, <span class="dt">nCores=</span><span class="dv">2</span>)</code></pre></div>
<p>The data frame <code>enrichment</code> now contains the enrichment informations between ENCFF001VCU and the ReMap catalogue given ENCFF718QVA as a universe.</p>
<p>For more permissive universe you can use the parameter included as the fraction of shuffled regions that must be at least contained in universe regions (1 by default). For example included = 0.1 would allow at least 10% of shuffled regions to be within the universe.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create the enrichment with a less restrictive universe.</span>
enrichment &lt;-<span class="st"> </span><span class="kw">enrichment</span>(ENCFF001VCU, remapCatalog, universe, <span class="dt">included =</span> <span class="fl">0.1</span>, <span class="dt">nCores=</span><span class="dv">2</span>)
<span class="co"># 90% of the shuffled regions can now be outside of the universe regions.</span></code></pre></div>
<p>Shuffling regions occurs in the whole genome but it is possible to restrict shuffles to occur only in the chromosome they originate with the parameter byChrom (<code>FALSE</code> by default).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create the enrichment with a less restrictive universe.</span>
enrichment &lt;-<span class="st"> </span><span class="kw">enrichment</span>(ENCFF001VCU, remapCatalog, universe, <span class="dt">included =</span> <span class="fl">0.1</span>, <span class="dt">byChrom =</span> <span class="ot">TRUE</span>, <span class="dt">nCores=</span><span class="dv">2</span>)
<span class="co"># 90% of the shuffled regions can now be outside of the universe regions.</span>
<span class="co"># The shuffled regions are still in the same chromosome where they came from.</span></code></pre></div>
</div>
</div>
<div id="shuffling-and-random-generation" class="section level1">
<h1>Shuffling and random generation</h1>
<p>As the most basic enrichment function uses random generations in order to estimate the p-values, shuffling functions and random generations are available in this package. In this two functions, it is possible to use all the parameters already mentioned (universe, included, byChrom).</p>
<div id="shuffling" class="section level2">
<h2>Shuffling</h2>
<p>Shuffling regions consists in randomly reordering the positions of the query regions within a genome.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Shuffling ENCFF001VCU</span>
shuffledENCFF001VCU &lt;-<span class="st"> </span><span class="kw">shuffle</span>(ENCFF001VCU, <span class="dt">universe =</span> universe, <span class="dt">byChrom =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="random-generations" class="section level2">
<h2>Random generations</h2>
<p>It is sometimes useful to generate purely random regions, for negatives controls for example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Generate 100 random regions with a size of 1000 bases pair.</span>
randomRegions &lt;-<span class="st"> </span><span class="kw">genRegions</span>(<span class="dv">100</span>, <span class="dv">1000</span>)</code></pre></div>
</div>
</div>
<div id="other-assemblies" class="section level1">
<h1>Other assemblies</h1>
<p>For now, only the hg38 assembly has been used by default. It is important to know how to make enrichment analysis with other genomes. All functions in the ReMapEnrich package that uses shuffles or random regions generation must ackowledge the sizes of the chromosomes of the species in consideration.</p>
<p>It is possible to load the chromosomes sizes of hg38 in one function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hg38ChromSizes &lt;-<span class="st"> </span><span class="kw">loadChromSizes</span>(<span class="st">&quot;hg38&quot;</span>)</code></pre></div>
<p>But you may want to download other assemblies from the UCSC database.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Example with rn5</span>
rn5ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;rn5&quot;</span>)

<span class="co"># Creation of random regions in the rattus norvegicus genome.</span>
randomRegions &lt;-<span class="st"> </span><span class="kw">genRegions</span>(<span class="dv">100</span>, <span class="dv">1000</span>, rn5ChromSizes)

<span class="co"># Shuffling of regions in the rattus norvegicus genome.</span>
shuffledRegions &lt;-<span class="st"> </span><span class="kw">shuffle</span>(randomRegions, rn5ChromSizes)

<span class="co"># Species relevant to current and future ReMap releases</span>
hg38ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;hg38&quot;</span>)
hg19ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;hg19&quot;</span>)
mm10ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;mm10&quot;</span>)
dm6ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;dm6&quot;</span>)

<span class="co"># Maybe one day</span>
ce11ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;ce11&quot;</span>)
rn5ChromSizes &lt;-<span class="st"> </span><span class="kw">downloadUcscChromSizes</span>(<span class="st">&quot;rn5&quot;</span>)

<span class="co">#-- This warning message is known an issue from RMySQL</span>
<span class="co"># https://github.com/r-dbi/RMySQL/issues/37</span>

<span class="co"># Warning message:</span>
<span class="co"># In .local(conn, statement, ...) :</span>
<span class="co">#  Unsigned INTEGER in col 1 imported as numeric</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Cheneby:2018ix">
<p>1. Chèneby J, Gheorghe M, Artufel M, Mathelier A, Ballester B. ReMap 2018: an updated atlas of regulatory regions from an integrative analysis of DNA-binding ChIP-seq experiments. Nucleic acids research. 2018;46:D267–75.</p>
</div>
<div id="ref-Griffon:2015en">
<p>2. Griffon A, Barbier Q, Dalino J, Helden J van, Spicuglia S, Ballester B. Integrative analysis of public ChIP-seq experiments reveals a complex multi-cell regulatory landscape. Nucleic acids research. 2015;43:e27–7.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
