---
title: "roken: Quick tour"
author: "Martin Mestdagh, Zacharie Ménétrier"
date: "`r Sys.Date()`"
package: "`r pkg_ver('roken')`"
vignette: >
    %\VignetteIndexEntry{roken-quick-tour}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
output: 
    BiocStyle::html_document
bibliography: 
    bibliography.bib
csl:
    biomed-central.csl
---

```{r, echo=FALSE}
knitr::opts_chunk$set(collapse = T, 
                      comment = "##    ", 
                      eval=TRUE, 
                      echo=FALSE, 
                      warning = FALSE, 
                      results = FALSE, 
                      message = FALSE)
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
```

# Abstract

Current next generation sequencing studies generate a large variety of genomic regions ranging from regulatory regions with transcription factors or histone marks ChIP-seq to variant calls or coding/non-coding transcripts. Also, the number of complex catalogues from large-scale integrative efforts are increasing[@Cheneby:2018ix; @Griffon:2015en; @Ashoor:2015ey; @Amin:2015jm] and large sequencing projects[@RoadmapEpigenomicsConsortium:2015in; @ENCODEProjectConsortium:2012gc; @Fernandez:2016do].  To facilitate the interpretation of functional genomics, epigenomics and genomics data we have developed a R-software package ‘roken’ to identify significantly enriched regions from user defined catalogues. Roken provide functions to import any in-house catalogue, automate and plot the enrichment analysis for genomic regions. 

# Quick example

This example is based on small datasets released with the roken package. 

It will go through the folowing steps. 

## Loading a minimal set of regions

- A **query set** of genomic regions (all the annotated SOX2 binidng regions on the chromosome 22).
- Load a (reduced) catalogue of reference genomic regions (all the annotated regions of the chromosome22). 


```{r init_query, echo=TRUE}
# Load the roken library
library(roken) 

# Load the example dataset
query <- bedToGranges(system.file("extdata",
                                  "ReMap_nrPeaks_public_chr22_SOX2.bed",
                                  package = "roken"))

catalog <- bedToGranges(system.file("extdata",
                                    "ReMap_nrPeaks_public_chr22.bed",
                                    package = "roken"))
```

We now dispose of two genomic ranges object: `query` and `catalog`.

The query and catalog are both sets of genomic ranges, defined by attributes such as chromosome, start, end, strand. 

Notes 

- ChIP-seq genomic ranges are strand-insensitive, and therefore this field is left undefined (*). 
- In ReMap features, "ID" column indicates the name of the transcriptional regulator name, where  "regulator" is a generic concept covering transcription factor, insulator, chromatin remodelling factor, .... 



```{r echo=TRUE, results=TRUE}
# Check what the catalog contains
print(catalog)

# Check what the query contains
print(query)

```

As you can see, it is important for the catalogue to have proper IDs for each regions. All IDs found in the catalogue will be taken as a category and will be computed in enrichment analysis.
The IDs of the query will not be used for the rest of the enrichment analysis and can be ignored.

## Computing enrichment of intersections between the query and each entry of the catalogue

```{r echo=TRUE, results=TRUE}
enrichment <- enrichment(query, catalog, byChrom = TRUE)
head(enrichment)
```

The option `byChrom` is set to TRUE as we are only working on one chromosome for this analysis.




## Format of the result

The data frame enrichment now contains all the features of the catalogue with their enrichment values.

1. **Category:**  identifier of the genomic features (regulators in our case) of the catalogue (ReMap here) showing significant overlap with the query. 

2. **Number of overlap:** number of overlaps between the query and each categories of the catalogue.

3. **Random average:**  mean number of overlaps between all the shuffles and the catalogue.

4. **Mapped peaks ratio:** percentage of covering of each categories of the catalogue by the query.

5. **Effect size:** log value of the number of overlaps divided by the random average.

6. **P-significance:** minus-logarithm of the p-value: $p_{sig} = -log_{10}(p)$, enabling to highlight the relevant order of magnitude on graphical representations (e.g. volcano plot). 

7. *P-value:* probability to observe an effect at least as extreme as the result, under null hypothesis, which can be interpreted as an estimation of the false positive rate (FPR). The p-value is computed with the Poisson distribution (and validated empirically with randomized query regions).

8. **Q-significance: ** minus-logarithm of the q-value: $q_{sig} = -log_{10}(q)$

9. **Q-value:** correction of the p-value to take into account the multiple testing (due to the fact that the query is compared to each regulator of the remap catalogue). The q-value is an estimation of the false discovery rate (FDR). Several multiple-testing correction methods are supported by roken (default: "BH" for Benjamini-Hochberg).

10. **E-significance:** minus-logarithm of the e-value: $q_{sig} = -log_{10}(E)$

11. **E-value:** expected number of false positives for a given p-value. 

## Statistical interpretation

The data frame is ordered by decreasing significance ("p.significance" column).


## Biological interpretation of the results

It is interesting to give some hints about the biological relevance of the results. 


This kind of enrichment analysis could be interpreted as a co-location detector, as it will give the regulators showing a significant co-location with the query (SOX2 peaks on chromsome 22) in the reference genome (hg19 assembly).

SOX2, is a transcription factor essential to maintain self-renewal and pluripotency of undifferentiated embryonic stem cells.

- Not surprisingly, the most significant overlap for our query is the catalogue entries for SOX2. 

- The second hit is NANOG, a transcription factor critically involved in self-renewal of undifferentiated embryonic stem cells. 

- The third hit is Pou5f1, which forms a heterodimer with Sox2 and therefore co-occurs in most of its peaks. 

Chen et al (2008) showedd that Nanog peaks strongly overlap with Sox2 and Pou5f (also called Oct4). 

It is important to specify that this example is only relevant for the hg19 assembly. If any other assemblies are used, the parameter "chromSizes" should be acknowledged accordingly




This vignette will presents graphical representations of enrichment analysis that are implemented in this package.

# Graphical functions

Multiple graphical representations of enrichment analysis are implemented in this package.

## Bar plot

We can now display a bar plot which is the most basic representation of an enrichment analysis.

```{r}
# Display a bar plot
enrichmentBarPlot(enrichment, sigDisplayQuantile = 0.5, top = 20, aRisk = 0.00001)
```

The displayed plot represents the most significant categories and their q-significances. The sigDisplayQuantile parameter allow to choose a quatile limit at which the bars will stop expanding. The values of those bars will be displayed on it.
The top parameter allow to choose how many categories you want to be displayed on the plot, and the aRisk represents the alpha risk that will be rescaled as the significance in order to see which categories reject the null hypothesis.

## Volcano plot

A volcano plot is a type of scatter-plot that is used to quickly identify changes in large data sets composed of replicate data. It plots significance versus fold-change on the y and x axes, respectively. As effect size in enrichment analysis is rarely negative you must expect the volcano plot to only expand in one direction.

```{r, echo=TRUE}
# Display a volcano plot (na.omit() is mandatory as there is NAs in the enrichment data frame).
enrichmentVolcanoPlot(na.omit(enrichment), sigDisplayQuantile = 0.9, aRisk = 0.00001)
```

## Dot plot

The enrichment dot plot is a good way to compare various informations in the enrichment analysis. It gives hints about the number of overlaps, the mapped peaks ratio and the q-significance for the given number of categories.

```{r, echo=TRUE, fig.height=7}
# Display a dot plot.
enrichmentDotPlot(enrichment)
```

It should be interesting here to avoid representation of the SOX2 category as it enriched against itself in the chromosome 22.

```{r, echo=TRUE, fig.height=7}
# Display a dot plot without SOX2.
enrichmentDotPlot(enrichment[enrichment$category != "SOX2",])
```


# References
