---
title: "roken: Quick tour"
author: "Martin Mestdagh, Zacharie Ménétrier"
date: "`r Sys.Date()`"
package: "`r pkg_ver('roken')`"
vignette: >
    %\VignetteIndexEntry{roken-quick-tour}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
output: 
    BiocStyle::html_document
bibliography: 
    bibliography.bib
csl:
    biomed-central.csl
---

```{r, echo=FALSE}
knitr::opts_chunk$set(collapse = T, comment = "#>", eval=TRUE)
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
```

# Abstract

Current next generation sequencing studies generate a large variety of genomic regions ranging from regulatory regions with transcription factors or histone marks ChIP-seq to variant calls or coding/non-coding transcripts. Also, the number of complex catalogues from large-scale integrative efforts are increasing[@Griffon:2015en; @Ashoor:2015ey; @Amin:2015jm] and large sequencing projects[@RoadmapEpigenomicsConsortium:2015in; @ENCODEProjectConsortium:2012gc; @Fernandez:2016do].  To facilitate the interpretation of functional genomics, epigenomics and genomics data we have developed a R-software package ‘roken’ to identify significantly enriched regions from user defined catalogues. Roken provide functions to import any in-house catalogue, automate and plot the enrichment analysis for genomic regions. 

# Quick example

This example is based on pet datasets released with the roken package. 

It will go through the folowing steps. 

1. Load a minimal example

    - a query set of genomic regions (all the SOX2 annotated regions on the chromosome 22)
    - (reduced) catalogue of reference genomic regions (all the annotated regions of the chromosome22). 

```{r, echo=FALSE}
# Load the roken library
library(roken) 

# Load the example dataset
query <- BedToGranges(system.file("extdata",
                                  "ReMap_nrPeaks_public_chr22_SOX2.bed",
                                  package = "roken"))

catalog <- BedToGranges(system.file("extdata",
                                    "ReMap_nrPeaks_public_chr22.bed",
                                    package = "roken"))
```

We now dispose of two genomic ranges object.
The catalogue contains features names.

```{r}
# Check what contains the catalog
catalog
# Check what contains the query
query
```

As you can see, it is important for the catalogue to have proper id for each regions. All ids found in the catalogue will be taken as a category and will be computed in enrichment analysis.
The ids of the query will not be taken in account for the rest of the enrichment analysis and can be ignored.

2. Compute the enrichment of intersections between the query and each entry of the catalogue.

The option byChrom is set to TRUE as we are only working on one chromosome for this analysis.

```{r}
enrichment <- GrEnrichment(query, catalog, byChrom = TRUE)
head(enrichment)
```

This kind of enrichment analysis could be interpretated as a co-location finder, as it will give the annotated functions that are co-located with SOX2 in the chromsome 22 for the hg19 assembly.
SOX2, is a transcription factor that is essential for maintaining self-renewal, or pluripotency, of undifferentiated embryonic stem cells.
NANOG is a transcription factor critically involved with self-renewal of undifferentiated embryonic stem cells.
The results are interesting as we can see that the NANOG transcription factor which is the most significant result for SOX2 (after itself) is also involved in undifferentiated embryonic stem cells.

The data frame enrichment now contains all the features of the catalogue with their enrichment values.
The column "p.significance" is the one ordering the data frame.
It is important to specify that this example is only relevant for the hg19 assembly.
If any other assembly is computed, the parameter "chromSizes" should be acknowledged consequently.

# Enrichment analysis
The data frame is composed of eleven columns with differents informations about the enrichment.

1. Category:
This column contains features names of each category that have been found in the catalogue.

2. Number of overlaps:
This column contains the number of overlaps between the query and each categories of the catalogue.

3. Random average:
This column contains the mean number of overlaps between all the shuffles and the catalogue.

4. Mapped peaks ratio:
The mapped peaks ratio represents the percentage of covering of each categories of the catalogue by the query.

5. Effect size:
The effect size is simply the log value of the number of overlaps divided by the random average.

6. P-significance:
The p-significance column is a rescaled value of the p-value, obtained by -log10(Pval).

7. P-value:
The p-value is the probability that random overlaps gived, at least, results as extremes as the number of overlaps following the Poisson distribution.

8. Q-significance:
The q-significance column is a rescaled value of the p-value, obtained by -log10(Qval).

9. Q-value:
The q-value is the corrected p-value given multiple comparisons testing. It is possible to choose the correction method (fdr by default).

10. E-significance:
The e-significance column is a rescaled value of the p-value, obtained by -log10(Eval).

11. E-value:
This last column contains the expected numbers of false discoveries for each categories.

This vignette will presents graphical representations of enrichment analysis that are implemented in this package.

# Graphical functions

Multiple graphical representations of enrichment analysis are implemented in this package.

## Bar plot

We can now display a bar plot which is the most basic representation of an enrichment analysis.

```{r}
# Display a bar plot
EnrichmentBarPlot(enrichment, sigDisplayQuantile = 0.5, top = 20, aRisk = 0.00001)
```

The displayed plot represents the most significant categories and their q-significances. The sigDisplayQuantile parameter allow to choose a quatile limit at which the bars will stop expanding. The values of those bars will be displayed on it.
The top parameter allow to choose how many categories you want to be displayed on the plot, and the aRisk represents the alpha risk that will be rescaled as the significance in order to see which categories reject the null hypothesis.

## Volcano plot

A volcano plot is a type of scatter-plot that is used to quickly identify changes in large data sets composed of replicate data. It plots significance versus fold-change on the y and x axes, respectively. As effect size in enrichment analysis is rarely negative you must expect the volcano plot to only expand in one direction.

```{r, echo=TRUE}
# Display a volcano plot (na.omit() is mandatory as there is NAs in the enrichment data frame).
EnrichmentVolcanoPlot(na.omit(enrichment), sigDisplayQuantile = 0.9, aRisk = 0.00001)
```

## Dot plot

The enrichment dot plot is a good way to compare various informations in the enrichment analysis. It gives hints about the number of overlaps, the mapped peaks ratio and the q-significance for the given number of categories.

```{r, echo=TRUE, fig.height=7}
# Display a dot plot.
EnrichmentDotPlot(enrichment)
```

It should be interesting here to avoid representation of the SOX2 category as it enriched against itself in the chromosome 22.

```{r, echo=TRUE, fig.height=7}
# Display a dot plot without SOX2.
EnrichmentDotPlot(enrichment[enrichment$category != "SOX2",])
```


# References
