---
title: "Roken: Genomic region enrichment analyses for large genomic catalogues"
author: "Vignette Author"
date: "`r Sys.Date()`"
package: "`r pkg_ver('roken')`"
vignette: >
    %\VignetteIndexEntry{Roken}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
output: 
    BiocStyle::html_document
bibliography: 
    bibliography.bib
csl:
    biomed-central.csl
---

```{r, echo = FALSE, message = FALSE}
knitr::opts_chunk$set(collapse = T, comment = "#>", eval=FALSE)
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
```

# Abstract

Current next generation sequencing studies generate a large variety of genomic regions ranging from regulatory regions with transcription factors or histone marks ChIP-seq to variant calls or coding/non-coding transcripts. Also, the number of complex catalogues from large-scale integrative efforts are increasing[@Griffon:2015en; @Ashoor:2015ey; @Amin:2015jm] and large sequencing projects[@RoadmapEpigenomicsConsortium:2015in; @ENCODEProjectConsortium:2012gc; @Fernandez:2016do].  To facilitate the interpretation of functional genomics, epigenomics and genomics data we have developed a R-software package ‘Roken’ to identify significantly enriched regions from user defined catalogues. Roken provide functions to import any in-house catalogue, automate and plot the enrichment analysis for genomic regions. 


# Quick example

This example is based on pet datasets released with the roken package. 

It will go through the folowing steps. 

1. Load a minimal example

    - a query set of genomic regions. 
    - (reduced) catalogue of reference genomic regions. 

2. Compute the enrichment of intersections between the query and each entry of the catalogue. 

```{r}
# Load the roken library
library(roken) 

# Load the example dataset
query <- BedToGranges(system.file(
    "extdata",                                    
    "ReMap_nrPeaks_public_chr22_SOX2.bed",
    package = "roken"))

catalog <- BedToGranges(system.file("extdata",
                                    "ReMap_nrPeaks_public_chr22.bed",
                                    package = "roken"))
```

We now dispose of two genomic ranges object.
The catalogue contains features names.

```{r}
enrichment <- GrEnrichment(query, catalog, shuffles = 10)
```

The data frame enrichment now contains all the features of the catalogue with their enrichment values.
The column "p.significance" is the one ordering the data frame.
It is important to specify that this example is only relevant for the hg19 assembly.
If any other assembly is computed, the parameter "chromSizes" should be acknowledged consequently.

## Catalogue loading 

### Downloading ReMap catalogue

```{r, echo=TRUE}
# Create a local directory for the tutorial
demo.dir <- "~/roken_demo"
dir.create(demo.dir, showWarnings = FALSE, recursive = TRUE)
setwd(demo.dir)
# Use the function DowloadRemapCatalog
remapCatalog <- DownloadRemapCatalog("nrpeaks_all.bed.gz")

```

The genomic ranges object remapCatalog now contains the full ReMap catalogue[@Griffon:2015en].
A file has also been downloaded at "~/roken_demo/nrpeaks_all.bed.gz" (0.544 GB when not compressed).

### Loading a custom catalogue




## Query region set loading

## Loading chromosome sizes
To compute enrichement we first need to load the genome file for the species and assembly of interest. This can be done with `LoadUcscChromSizes` using standard UCSC short names for assemblies (eg. hg19, hg38, mm10, mm9, rn5, etc..). 
```{r}
# This call will not create any file but only load the data in a vector.
hg19ChromSizes <- DownloadUcscChromSizes("hg19")

```
```{r}
head(hg19ChromSizes)

```
The vector hg19ChromSizes now contains all the chromosomes for the Homo sapiens NCBI37 (hg19) and their respective lengths.
A file has also been downloaded at "~/roken_demo/hg19_chrom_sizes.txt".


Alterntivaly, one can directly load a genome file from a file. 
```{r}
# It is also possible to load the chromosomes sizes of any species from a file.
hg19ChromSizes <- LoadChromFile("hg19")

```



## Negative control

A negative control consists in submitting datasets known to be under the null hypothesis. One approach is to pick up at random genomic regions of the same sizes as the query set. In principle the software should return no significant overlap. 

```{r}
## Generate a randomized query set
shuffledQuery <- GrShuffle(query)
shuffledEnrichment <- GrEnrichment(shuffledQuery, catalog)
```


# Displaying the results
enrichment <- na.omit(enrichment)
## P-value histogram

```{r}
par(mfrow = c(2,1))
hist(enrichment$p.value, breaks=20)
hist(shuffledEnrichment$p.value, breaks=20)
par(mfrow = c(1,1))
```

## Volcano plot

```{r}
EnrichmentVolcanoPlot(enrichment)
```

## Bar plot

```{r}
EnrichmentBarPlot(enrichment)
```

## Dot plot 

```{r}
EnrichmentDotPlot(enrichment)
```

# Example using ENCODE data as query

In this example we are using some H3K27ac ChIP-seq histone mark from ENCODE as a query against the ReMap catalogue. H3K27ac is associated with the higher activation of transcription and therefore defined as an active enhancer mark. H3K27ac is found at both proximal and distal regions of transcription start site (TSS). 


ENCFF001SUX H1-hESC
ENCFF575WAS HCT116
ENCFF023EER Right_lobe_liver
ENCFF826VBA Lung

```{r}
dat<-data.frame(encode=c("ENCFF001SUX","ENCFF575WAS", "ENCFF023EER", "ENCFF826VBA"), 
                assay=c("H1-hESC","HCT116","Right_lobe_liver","Lung") )
head(dat) 
```





# Example using CODEX and DENdb as catalogues 
Both CODEX and DENdb can be used as cataloguse just like ReMap as been used in the step-by-step quick example. 

CODEX [@SanchezCastillo:2015ev] provides access to processed and curated NGS experiments, including ChIP-Seq (transcription factors and histones), RNA-Seq and DNase-Seq. [NOTE: J'ai pas trouvé comment telecharger CODEX en fichier texte - a creuser - sinon a virer]

DENdb [@Ashoor:2015ey] is a repository of predicted enhancers derived from multiple human cell-lines. DENdb integrates enhancers predicted by five different methods (ChromHMM, Segway, RFECS, CSI-ANN, and ENCODE integrated annotation)
[Pas de format BED, faut importer ca en GenomicRange]

http://www.cbrc.kaust.edu.sa/dendb/src/ChIP_seq_TF.csv.zip


# Example CTCF peaks enriched in repeats



# Example using GTEx data as catlogue ??
Voir si on peut correler des regions (enhancers d'un tissue d'encode) avec l'expression des genes issu de GTEx. 
Je suis pas sur du truc a 100%
http://www.gtexportal.org/home/tissueSummaryPage


# Homepage
Please visit 'Roken homepage' for more informations and demo.

# Session Info
Here is the output of `sessionInfo()` on the system on which this document was compiled:
```{r, include=FALSE}
sessionInfo() 
```

# References


